### ISSUE TITLE
after use macro let,  shrinking don't work

### ISSUE THREAD

**Issue by apadryta** [alfert/propcheck#71]
when ledddt is use in larger proptest, the shrinking stoped working.
Here is small example to reconstruct this situation: 
```Elixir
defmodule PropTest.Equal do
  use ExUnit.Case
  use PropCheck
  use PropCheck.StateM.DSL

  def initial_state(), do: %{}

  defcommand :equal do
    def impl(number), do: :ok 
    def args(_state) do
      let ([num <-integer(1,1000)]) do
         [num]
      end
    end
    def post(_state, [arg], :ok), do: arg != 800 
    def next(model, [_arg], _ret), do: model
  end

  property "quick test of property test", [numtests: 1000] do
    forall cmds <- commands(__MODULE__) do
        events = run_commands(cmds)
        (events.result == :ok)
        |> when_fail(
            IO.puts """
            History: #{inspect events.history, pretty: true}
            State: #{inspect events.state, pretty: true}
            Env: #{inspect events.env, pretty: true}
            Result: #{inspect events.result, pretty: true}
            """)
        |> aggregate(command_names cmds)
    end
  end
end
```

**Comment by alfert** [comment]
Yes, this is a failure. I added a few simpler properties, where `let` is used and shrinking works. But in this simple DSL example it does not shrink. Exchanging the `let` by `integer(1, 1000)` shows a proper shrinking behaviour and finds the initial call with argument `800` after 50+ shrinking steps. 

I need to investigate whether this depends on the DSL.

**Comment by alfert** [comment]
Resolved in #76 Another generation algorithm was required. Thanks for the bug report.
